packages/vue/src/components/SpAssets/SpAssets.vue

```diff
@@ -57,26 +177,155 @@ export default defineComponent({
       )
     })
 
+    let isShowMore = computed(() => {
+      if (!balances.value.length) {
+        return
+      }
+
+      if (!state.searchQuery) {
+        return filteredBalanceList.value.length < balances.value.concat(state.demoDenoms).length
+      } else {
+        return filteredBalanceList.value.length < filteredArrayLength
+      }
+    })
+
+    const filteredBalanceList = computed(() => {
+      let searchArray = balances.value.length ? balances.value.concat(state.demoDenoms) : [],
+          searchString = state.searchQuery
+
+      if (!searchString) {
+        return searchArray.slice(0, state.displayLimit);
+      }
+
+      searchString = searchString.trim().toLowerCase()
+
+      searchArray = searchArray.filter((item) => {
+        if (item.denom.toLowerCase().includes(searchString)) {
+          return item
+        }
+      })
+      filteredArrayLength = searchArray.length
```

Comment:
why do we need this?

`isShowMore` and `filteredBalanceList` could use some simplification

Reply:
This logic is a bit confusing, I agree. If you have any ideas how it can be simplified then I will gladly do it.